File: create_symlinks.py
#!/usr/bin/env python

# npm will not install symlinks that are part of the git repository,
# so create them manually as a pre-install step instead

from subprocess import call
import os

def create_symlink(link_to, file_path):
    print("LOOK AT ME");
    call(["pwd"]);
    call(["find", "../.."])
    print('Creating symlink {} -> {}'.format(file_path, link_to))
    os.symlink(link_to, file_path)

create_symlink('libglapi.so.0.0.0', 'mesa/lib/libglapi.so.0')
create_symlink('libglapi.so.0.0.0', 'mesa/lib/libglapi.so')

create_symlink('libOSMesa.so.8.0.0', 'mesa/lib/libOSMesa.so.8')
create_symlink('libOSMesa.so.8.0.0', 'mesa/lib/libOSMesa.so')

File: create_symlinks.js
var fs = require('fs');

function create_symlink(srcpath, dstpath) {
  fs.symlink(srcpath, dstpath, function(err){
    console.log('Symlink', srcpath, dstpath);
    if (err) console.log('error', err);
  });
}

create_symlink('libglapi.so.0.0.0', 'mesa/lib/libglapi.so.0');
create_symlink('libglapi.so.0.0.0', 'mesa/lib/libglapi.so');

create_symlink('libOSMesa.so.8.0.0', 'mesa/lib/libOSMesa.so.8');
create_symlink('libOSMesa.so.8.0.0', 'mesa/lib/libOSMesa.so');